src/main/java/org/apache/sysml/hops/AggBinaryOp.java

```diff
@@ -115,7 +115,13 @@ public AggBinaryOp(String l, DataType dt, ValueType vt, OpOp2 innOp,
 		//compute unknown dims and nnz
 		refreshSizeInformation();
 	}
-	
+
+	@Override
+	public void checkArity() throws HopsException {
+		int sz = _input.size();
+		HopsException.check(sz == 2, this, "should have arity 2 but has arity %d", sz);
```

Comment:
Could you please substitute sz directly (whenever feasible) to increase readability?